endpoints/openrtb2/auction.go

```diff
@@ -1141,9 +1139,18 @@ func checkSafari(r *http.Request) (isSafari bool) {
 }
 
 // Write(return) errors to the client, if any. Returns true if errors were found.
-func writeError(errs []error, w http.ResponseWriter) bool {
+func writeError(errs []error, w http.ResponseWriter, labels *pbsmetrics.Labels) bool {
 	if len(errs) > 0 {
-		w.WriteHeader(http.StatusBadRequest)
+		httpStatus := http.StatusBadRequest
+		metricsStatus := pbsmetrics.RequestStatusBadInput
+		for _, err := range errs {
+			erVal := errortypes.DecodeError(err)
+			if erVal == errortypes.BlacklistedAppCode || erVal == errortypes.BlacklistedAcctCode {
+				httpStatus = http.StatusServiceUnavailable
+			}
+		}
+		w.WriteHeader(httpStatus)
+		labels.RequestStatus = metricsStatus
```

Comment:
Can anything change `metricsStatus` after line 1144? If not, why not combine the two lines and get rid of `metricsStatus`?

Reply:
I left it like that because in PR#1030 the blacklist metrics get implemented so the for loop changes to:
```
func writeError(errs []error, w http.ResponseWriter, labels *pbsmetrics.Labels) bool {
	if len(errs) > 0 {
		httpStatus := http.StatusBadRequest
		metricsStatus := pbsmetrics.RequestStatusBadInput
		for _, err := range errs {
			erVal := errortypes.DecodeError(err)
			if erVal == errortypes.BlacklistedAppCode || erVal == errortypes.BlacklistedAcctCode {
				httpStatus = http.StatusServiceUnavailable
   +	        metricsStatus := pbsmetrics.RequestBlacklisted
			}
		}
		w.WriteHeader(httpStatus)
		labels.RequestStatus = metricsStatus
		for _, err := range errs {
			w.Write([]byte(fmt.Sprintf("Invalid request: %s\n", err.Error())))
		}
		return true
	}
	return false
}
```
But I can change it real quick if you want

Reply:
Changed

Reply:
```
func writeError(errs []error, w http.ResponseWriter, labels *pbsmetrics.Labels) bool {
	if len(errs) > 0 {
		httpStatus := http.StatusBadRequest
		for _, err := range errs {
			erVal := errortypes.DecodeError(err)
			if erVal == errortypes.BlacklistedAppCode || erVal == errortypes.BlacklistedAcctCode {
				httpStatus = http.StatusServiceUnavailable
			}
		}
		w.WriteHeader(httpStatus)
		labels.RequestStatus = pbsmetrics.RequestStatusBadInput
		for _, err := range errs {
			w.Write([]byte(fmt.Sprintf("Invalid request: %s\n", err.Error())))
		}
		return true
	}
	return false
}
```